skada/utils.py

```diff
@@ -247,3 +253,219 @@ def source_target_split(
         (a[source_idx], a[~source_idx]) if a is not None else (None, None)
         for a in arrays
     ))
+
+
+def source_target_merge(
+    *arrays,
+    sample_domain : Optional[np.ndarray] = None
+) -> np.ndarray:
```

Comment:
Missed this one last time: the return type should be a tuple of ndarrays, moreover, of unknown size. `Sequence` would be probably the closest container type for such situations.